File: 1894-merge-strings-alternately/1894-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String w1, String w2) {
        
        int m= w1.length(); 
        int n= w2.length(); 
        int i=0,j=0; 
        StringBuilder sb= new StringBuilder(); 
        while(i<m && j<n ){

            sb.append(w1.charAt(i)); 
            sb.append(w2.charAt(j));
            i++; j++;

        }

        // if remaining 
        while(i<m)
       {     sb.append(w1.charAt(i)); 
            i++;}
   while(j<n)
       {     sb.append(w2.charAt(j)); 
            j++;}
        return sb.toString(); 
        
    }
}

File: 1894-merge-strings-alternately/1894-merge-strings-alternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        res=""
        m= len(word1)
        n= len(word2)
        i=0
        j=0

        while i<m and j<n: 
            res= res+ word1[i]
            res= res+word2[j]
            i+=1
            j+=1
            
        
        # remaining chars from word1 
        while i!=m : 
            res= res+ word1[i]
            i+=1
        while j!=n: 
            res= res+ word2[j]
            j+=1

        return res
        